### PR TITLE
Add point cloud gridding

### LINKED ISSUE #552: Add point cloud gridding into raster with nodata value setting
Right now we only have raster creation from a point cloud using `from_pointcloud_regular` (see #547), but for the generic case of any point cloud (irregular coordinates), we should have a gridding function based on `scipy.interpolate.griddata`.

### PR DESCRIPTION
**Pull request by rhugonnet**
Resolves #552

### CODE REVIEW

**Comment by rhugonnet** [comment]
Will be wrapped and documented in #492

File: geoutils/pointcloud.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import numpy as np
+from scipy.interpolate import griddata
+
+from geoutils._typing import NDArrayNum
+
+
+def _grid_pointcloud(
+    pc: gpd.GeoDataFrame,
+    grid_coords: tuple[NDArrayNum, NDArrayNum],
+    data_column_name: str = "b1",
```
Comment (adehecq) [inline geoutils/pointcloud.py:16]: Why is the default "b1"? Should it be None instead?

File: tests/test_pointcloud.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        # 2.1/ Grid points outside the convex hull of all points should always be nodata
+
+        # We convert the full raster to a point cloud, keeping all cells even nodata
+        rst_pc = rst.to_pointcloud(skip_nodata=False).ds
+
+        # We define a multi-point geometry from the individual points, and compute its convex hull
+        poly = geometry.MultiPoint([[p.x, p.y] for p in pc.geometry])
+        chull = poly.convex_hull
+
+        # We compute the index of grid cells interesting the convex hull
```
Comment (adehecq) [inline tests/test_pointcloud.py:49]: interesting -> intersecting

**Review by adehecq — COMMENTED** [review]
Amazing !! Nice functionality to have!

File: geoutils/pointcloud.py (hunk 1 above)
Comment (rhugonnet) [reply geoutils/pointcloud.py:16]: It's the same default of "band1" as for the `to_pointcloud()` function and `PointCloud()` class, but it's true it doesn't need a name here (like when creating a `gpd.GeoDataFrame`), so I guess if the default could be "None" and choose the first band by default...
Need to think a bit more on it... We can still adjust this during the `PointCloud` PR + release :wink:

File: geoutils/pointcloud.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    )
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=UserWarning, message="Geometry is in a geographic CRS.*")
+        near = gpd.sjoin_nearest(grid_pc, pc)
+        # In case there are several points at the same distance, it doesn't matter which one is used to compute the
+        # distance, so we keep the first index of closest point
+        index_right = near.groupby(by=near.index)["index_right"].min()
+
+    # Compute distance between points as a function of the pixel sizes in X and Y
+    res_x = np.abs(grid_coords[0][1] - grid_coords[0][0])
```
[the remark this answers is not in the document]
Comment (rhugonnet) [reply geoutils/pointcloud.py:64]: Added that it is always regular grid in the argument description (is also already mentioned in the first line of docstring).

File: tests/test_pointcloud.py (hunk 2 above)
Comment (rhugonnet) [reply tests/test_pointcloud.py:49]: Good catch!

File: geoutils/pointcloud.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rhugonnet) [reply geoutils/pointcloud.py:64]: And adding input check!

**Pull request merged**